src/modifiedsphericalbessel.jl

```diff
@@ -0,0 +1,36 @@
+
+"""
+    sphericalbesselk(nu, x::T) where T <: {Float32, Float64}
+
+Computes `k_{ν}(x)`, the modified second-kind spherical Bessel function, and offers special branches for integer orders.
+"""
+sphericalbesselk(nu, x) = _sphericalbesselk(nu, float(x))
+
+function _sphericalbesselk(nu, x::T) where T
+    if isinteger(nu) && nu < 41.5
+        # using ifelse here to hopefully cut out a branch on nu < 0 or not. The
+        # symmetry here is that 
+        # k_{-n} = (...)*K_{-n     + 1/2}
+        #        = (...)*K_{|n|    - 1/2}
+        #        = (...)*K_{|n|-1  + 1/2}
+        #        = k_{|n|-1}  
+        _nu = ifelse(nu<zero(nu), -one(nu)-nu, nu)
```

Comment:
What do we want to do for negative arguments? Do you need these routines for complex arguments? We should probably throw domain error for now.

Reply:
Argh, you're right. Apologies, I'm just not in the headspace of implementing this as a full function that will get called on its own and stuff. I don't need this for complex args and I'd be happy to put in the same check/error you use on `besselk`.

Reply:
No problem! Does NaN propagate through this as well? But I think the general structure of besselk for this is good. And we can add complex methods as we get to them.

Reply:
Another good catch. Just pushed something up that I think addresses both, although yet another branch due to an `isnan` to avoid just getting the domain error.